pkg/k8sops/statefulset.go

```diff
@@ -89,6 +90,27 @@ func NewBaseStatefulSet(ssName, isolationGroup string, cluster *myspec.M3DBClust
 		},
 	}
 
+	// Add SYS_RESOURCE security capability if not set (required to raise
+	// rlimit nofile from the process in container)
+	specSecurityCtx := cluster.Spec.SecurityContext
+	if specSecurityCtx == nil {
+		specSecurityCtx = &v1.SecurityContext{}
+	}
+	if specSecurityCtx.Capabilities == nil {
+		specSecurityCtx.Capabilities = &v1.Capabilities{}
+	}
+	hasCapabilitySysResource := false
+	for _, c := range specSecurityCtx.Capabilities.Add {
+		if c == capabilitySysResource {
+			hasCapabilitySysResource = true
+			break
+		}
+	}
+	if !hasCapabilitySysResource {
+		specSecurityCtx.Capabilities.Add =
```

Comment:
We'll have to only add this if the user hasn't set a securitycontext at all. If they've set one we can't modify it, as they may have [PodSecurityPolicies](https://kubernetes.io/docs/concepts/policy/pod-security-policy/) that prevent capabilities from being added, in which case they'd have no way to use the operator. Or we can add another flag to not add it.

I personally like only setting it if there's not an SC already